src/logic/contracts/methodIds.ts

```diff
@@ -1,5 +1,7 @@
 import { web3ReadOnly as web3 } from 'src/logic/wallets/getWeb3'
-import { DataDecoded, METHOD_TO_ID } from 'src/routes/safe/store/models/types/transactions.d'
+import { DataDecoded, METHOD_TO_ID, SAFE_METHODS_NAMES } from 'src/routes/safe/store/models/types/transactions.d'
+
+export type SafeMethods = typeof SAFE_METHODS_NAMES[keyof typeof SAFE_METHODS_NAMES]
```

Comment:
This is resolving to string, because `SAFE_METHODS_NAMES` is to defined as a const.
 I would suggest these changes:
 1. add `as const` to `SAFE_METHODS_NAMES`
 2. change this type like so:
 `type SafeMethodsKeys = keyof typeof SAFE_METHODS_NAMES`;
 `type SafeMethodsValues = typeof KeyToVal[SafeMethodsKeys]`;
 3. move the types from (2) to the same file where `SAFE_METHODS_NAMES` is defined.

Reply:
Guys, while doing the refactor I found this code:
```
const CollectibleSelectField = ({ initialValue, tokens }) => {
  const classes = useCollectibleSelectFieldStyles()

  return (
    <Field
      className={classes.selectMenu}
      component={SelectField}
      disabled={!tokens.length}
      initialValue={initialValue}
      name="nftTokenId"
      renderValue={(nftTokenId) => <SelectedCollectible tokenId={nftTokenId} tokens={tokens} />}
      validate={required}
    >
      {tokens.map((token) => (
        <MenuItem key={`${token.assetAddress}-${token.tokenId}`} value={token.tokenId}>
          <ListItemIcon className={classes.tokenImage}>
            <TokenLogo height={28} tokenAddress={token.address} />
          </ListItemIcon>
          <ListItemText primary={token.name} secondary={`token ID: ${token.tokenId}`} />
        </MenuItem>
      ))}
    </Field>
  )
}
```

Please check this line: 
```
{tokens.map((token) => (
        <MenuItem key={`${token.assetAddress}-${token.tokenId}`} value={token.tokenId}>
          <ListItemIcon className={classes.tokenImage}>
            <TokenLogo height={28} tokenAddress={token.address} />
          </ListItemIcon>
          <ListItemText primary={token.name} secondary={`token ID: ${token.tokenId}`} />
        </MenuItem>
      ))}
```

We cannot use `useToken` hook there, we were just passing the address to the `TokenLogo` component and let him do it on his own, what do you suggest to do there?

Reply:
I didn't get this, could you please give me an example?